lib/relation-definition.js

```diff
@@ -1573,12 +1614,13 @@ RelationDefinition.hasAndBelongsToMany = function hasAndBelongsToMany(modelFrom,
   options.options = params.options;
 
   if (params.polymorphic) {
-    var polymorphic = polymorphicParams(params.polymorphic);
+    var relationName = params.as || i8n.camelize(modelTo.pluralModelName, true);
```

Comment:
`params.reference || params.as || i8n.camelize(modelTo.pluralModelName, true);` if we decide to use `reference`